### DISCUSSION TITLE
index.html not working

### DISCUSSION THREAD

**Discussion by ghost** [community/community#23625] [Other Feature Feedback, Questions, & Ideas]
<p>I have provided an index.html file in my github pages site, however when I try to access it with a root url it throws a 404. When I add /index.html to the end it works perfectly. You can see this happening in the attached image </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/2bda860c09bf74ab710399a2566099b7abc43ef5.png" data-download-href="/uploads/short-url/6fWIo1NqvHgv2M8fYrQ9hCHseot.png?dl=1" title="b7b6dc5e1a3889a09e0d53088f55b25c" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10137/181078314-c7478448-7c41-43ac-aa0a-e1103445b15f.png" alt="b7b6dc5e1a3889a09e0d53088f55b25c" data-base62-sha1="6fWIo1NqvHgv2M8fYrQ9hCHseot" width="690" height="413" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/2bda860c09bf74ab710399a2566099b7abc43ef5_2_690x413.png, https://aws1.discourse-cdn.com/github/optimized/1X/2bda860c09bf74ab710399a2566099b7abc43ef5_2_1035x619.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/2bda860c09bf74ab710399a2566099b7abc43ef5_2_1380x826.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/2bda860c09bf74ab710399a2566099b7abc43ef5_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">b7b6dc5e1a3889a09e0d53088f55b25c</span><span class="informations">2878×1724 190 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/0d9d0ba1ded8d80a6f58e252457e23ea80d42665.png" data-download-href="/uploads/short-url/1WqFSiEJIrCnA36eMpvV9cdEpWB.png?dl=1" title="30e9fbd8bcc10bbd1403ace687e66686" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10137/181078315-09832699-0327-4117-9896-0d9855e7513c.png" alt="30e9fbd8bcc10bbd1403ace687e66686" data-base62-sha1="1WqFSiEJIrCnA36eMpvV9cdEpWB" width="690" height="412" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/0d9d0ba1ded8d80a6f58e252457e23ea80d42665_2_690x412.png, https://aws1.discourse-cdn.com/github/optimized/1X/0d9d0ba1ded8d80a6f58e252457e23ea80d42665_2_1035x618.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/0d9d0ba1ded8d80a6f58e252457e23ea80d42665_2_1380x824.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/0d9d0ba1ded8d80a6f58e252457e23ea80d42665_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">30e9fbd8bcc10bbd1403ace687e66686</span><span class="informations">2880×1720 359 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>.
<p>The repository running the github site is at <a href="https://github.com/NicksWorld/Codes-js-demo" rel="nofollow noopener">https://github.com/NicksWorld/Codes-js-demo</a> .</p>

**Comment by winstliu** [comment]
<p>Hmm, I can’t reproduce this - navigating to your website without index.html works fine.  Can you try again now?  And if still doesn’t work, what browser version are you using?</p>

**Comment by ghost** [comment, marked as answer]
<p>I already restructured the setup of the site, now it is all in the user site instead of project, and I use firefox.</p>

**Reply by billybobjones0701** [reply]
Ehat i dont understand what you meant

**Comment by gpuma** [comment]
<p>I had this exact error, my index.html was just a plaintext:</p>
<pre><code>hola amigos
</code></pre>
<p>Then I changed it to a proper HTML doc:</p>
<pre><code>&lt;!DOCTYPE html&gt;
&lt;html&gt;
&lt;head&gt;
  &lt;meta charset="UTF-8"&gt;
  &lt;title&gt;Página de prueba&lt;title&gt;
&lt;/head&gt;
&lt;body&gt;
  &lt;h1&gt;Hola amiguitos!&lt;/h1&gt;
  &lt;footer&gt;gpuma 2017&lt;/footer&gt;
&lt;/body&gt;
&lt;/html&gt;
</code></pre>
<p> I commited the new changes and it worked fine now.</p>

**Comment by bkwesi** [comment]
<p>Hmm, I’m a new developer, but I have the same issue, however, everything mentioned above in terms of &lt;!DOCTYPE html&gt; was already implemented. </p>
<p>I committed almost 21 hours ago and still nothing. With index.html at the end everything works. I see each product and can click into each product page.  </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/d44a929067e98340e6f68035e640c199ff9348dc.jpeg" data-download-href="/uploads/short-url/ui0ZFijv1S9WzgGW6AYWBubAwyg.jpg?dl=1" title="Screen Shot 2018-04-28 at 2.24.11 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/22344275/181078319-80a3bdc3-1749-4c11-a9e8-238417cb84ef.jpg" alt="Screen Shot 2018-04-28 at 2.24.11 PM.png" data-base62-sha1="ui0ZFijv1S9WzgGW6AYWBubAwyg" width="690" height="414" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/d44a929067e98340e6f68035e640c199ff9348dc_2_690x414.jpg, https://aws1.discourse-cdn.com/github/optimized/1X/d44a929067e98340e6f68035e640c199ff9348dc_2_1035x621.jpg 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/d44a929067e98340e6f68035e640c199ff9348dc_2_1380x828.jpg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/d44a929067e98340e6f68035e640c199ff9348dc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2018-04-28 at 2.24.11 PM.png</span><span class="informations">2812×1688 486 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>Without index.html products are missing and only one of the products (The 1st one) redirects to its product page. <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/6a4594b0d77aa200a6be4a118001419d90b18164.jpeg" data-download-href="/uploads/short-url/fa7GF26P6eF2kVBlPmOuwE0PxPu.jpg?dl=1" title="Screen Shot 2018-04-28 at 2.24.20 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/22344275/181078320-892b4b1d-5175-435b-a522-9408cb155271.jpg" alt="Screen Shot 2018-04-28 at 2.24.20 PM.png" data-base62-sha1="fa7GF26P6eF2kVBlPmOuwE0PxPu" width="690" height="413" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/6a4594b0d77aa200a6be4a118001419d90b18164_2_690x413.jpg, https://aws1.discourse-cdn.com/github/optimized/1X/6a4594b0d77aa200a6be4a118001419d90b18164_2_1035x619.jpg 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/6a4594b0d77aa200a6be4a118001419d90b18164_2_1380x826.jpg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/6a4594b0d77aa200a6be4a118001419d90b18164_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2018-04-28 at 2.24.20 PM.png</span><span class="informations">2806×1680 263 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by vikrantdhimate** [comment]
<p>Thanks a lot. It’s working as expected after this changes. Thanx.</p>

**Comment by Aakash1393** [comment]
<p>Can you please elaborate the solution as I am facing same problem</p>

**Comment by SerhiiKozachenko** [comment]
<p>I have added front matter part to the readme.md and it fixed the problem.</p>
<pre><code>---
permalink: /index.html
---
</code></pre>
<p>After this it shows index.html. I have found this solution also helps for custom 404 pages. <a href="https://help.github.com/articles/creating-a-custom-404-page-for-your-github-pages-site/" rel="nofollow noopener">https://help.github.com/articles/creating-a-custom-404-page-for-your-github-pages-site/</a></p>

**Comment by ikartiks** [comment]
<p>How can you tell us?</p>

**Comment by ashieshwatts** [comment]
<p>suffering from same problem how do i restructured the setups ?</p>

**Comment by fotisj** [comment]
<p>I have the impression this is a workaround but not a solution. If you access your site via <a href="http://github.com" rel="nofollow noopener">github.com</a>, you want the content of readme.md displayed, so people know, what this project is about. If people access your website via <a href="http://github.io" rel="nofollow noopener">github.io</a>, you want the content of index.html displayed. I was under the impression, that the right way to do this, is to exclude README.md from your website by including in your _config.yml:</p>
<p>exclude:</p>
<p>    - README.md</p>
<p>Now the problems seems to be, that this is not always working as it should.</p>

**Comment by mikkel250** [comment]
<p>FWIW, for those (other) newbs out there, I found that if you have an existing repo, what they don’t mention in the GH Pages tutorial, is you just need to to go to settings in the repo, then scroll down to the Github Pages section, and then set the (master) branch (this is disabled by default), and save it.</p>
<p>This should create a page you can then goto in the browser. </p>
<p> EDIT: theme is optional - I left it off. </p>
<p>See photos for visual guide. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f04c36ec44acfb6f65d3de6f2e390313f660555b.jpeg" data-download-href="/uploads/short-url/yhLSFGzGxXKTXEu0xXDzd3t88Cn.jpg?dl=1" title="settings github.JPG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/11878116/181078323-fb32cdcb-cbea-4c47-80a1-9d4deb8a0bfc.jpg" alt="settings github.JPG" data-base62-sha1="yhLSFGzGxXKTXEu0xXDzd3t88Cn" width="690" height="394" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/f04c36ec44acfb6f65d3de6f2e390313f660555b_2_690x394.jpg, https://aws1.discourse-cdn.com/github/optimized/2X/f/f04c36ec44acfb6f65d3de6f2e390313f660555b_2_1035x591.jpg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/f/f04c36ec44acfb6f65d3de6f2e390313f660555b.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f04c36ec44acfb6f65d3de6f2e390313f660555b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">settings github.JPG</span><span class="informations">1259×719 71.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0ca1c07609d8f50594041b2bd74f40b12a8f1525.jpeg" data-download-href="/uploads/short-url/1NKhrf8Pz3G5jiQTKHOwomoHFl3.jpg?dl=1" title="set branch.JPG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/11878116/181078324-d5832e9e-6e0a-467c-a028-4708610d4f11.jpeg" alt="set branch.JPG" data-base62-sha1="1NKhrf8Pz3G5jiQTKHOwomoHFl3" width="690" height="321" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0ca1c07609d8f50594041b2bd74f40b12a8f1525_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">set branch.JPG</span><span class="informations">1120×522 44.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Ispirett** [comment]
<p>thanks this worked great!</p>

**Comment by Yolanda-Gao** [comment]
<p>Thanks! This solved my problem! </p>

**Comment by DanMayer218** [comment]
<p>Thanks for the tip with the permalink.  It worked right away.</p>

**Reply by NikitaClarigo** [reply]
how you use permalink ? will you please show us by screenshot .

**Comment by Louvivien** [comment]
<p>I had the same problem and it was because I had an empty line before&lt;!<em>doctype html</em>&gt; in my index.html</p>

**Comment by matthewmain** [comment]
<p>Thanks Louvivien! That was my issue as well–never would have solved it without your tip.</p>

**Comment by andyneff** [comment]
<p>I just had this exact same problem, but AFTER it was working</p>
<p>It happened because I had to reset the gh-pages branch. So I</p>
<ol>
<li>git branch -D gh-pages</li>
<li>git push -d origin gh-pages</li>
</ol>
<p>Then I had a script create an orphan branch gh-pages, populate it, and push it. Before this, everything worked without /index.html, but after, I had to add /index.html. I must have triggered a bug by accident.</p>
<p>The fix was was go into the settings and change the GitHub Pages Source branch to master, then back to gh-pages. This toggle was able to get around the bugged state I found myself in and everything is good again.</p>

**Comment by six6out** [comment]
<p><strong>This is not related to GitHub pages.</strong></p>
<p>Wordpress is not assosiated with GitHub pages or GitHub in any way, except for being open source.</p>
